janitor/timeseries.py

```diff
@@ -176,3 +180,270 @@ def sort_timestamps_monotonically(
+            2. percentage (consider the percentage change between rows).
+        Defaults to percentage.
+    :param direction: Type of method used to handle the sign change when
+        comparing consecutive rows.
+        Acceptable arguments are:
+            1. increasing (only consider rows that are increasing in value).
+            2. decreasing (only consider rows that are decreasing in value).
+            3. any (consider rows that are either increasing or decreasing;
+                sign is ignored).
+        Defaults to any.
+    :param threshold: The value to check if consecutive row comparisons
+        exceed. Always uses a greater than comparison. Must be >= 0.0.
+        Defaults to 0.0
+    :param strict: flag to enable/disable appending of a flag column for
+        each column in the provided dataframe.
+        If set to True, will only append a flag column for those columns
+            found in at least one of the input dictionaries.
+        If set to False, will append a flag column for each column found
+             in the provided dataframe. If criteria is not specified,
+             the defaults for each criteria is used.
+        Defaults to False.
+    :returns: Dataframe that has flag jump columns.
+    :raises: JanitorError if ``strict=True`` and at least one of
+        ``scale``, ``direction``, or ``threshold`` inputs is not a
+        dictionary.
+    :raises: JanitorError if ``scale`` is not one of
+        ``["absolute", "percentage"]``.
+    :raises: JanitorError if ``direction is not one of
+        ``["increasing", "decreasing", "any"]``.
+    :raises: JanitorError if ``threshold`` is less than 0.0.
+    """
+    df = df.copy()
+
+    if strict:
+        if (
+            any(isinstance(arg, dict) for arg in (scale, direction, threshold))
+            is False
+        ):
+            raise JanitorError(
+                "When enacting 'strict=True', 'scale', 'direction', or "
+                + "'threshold' must be a dictionary."
+            )
+
+        # Only append a flag col for the cols that appear
+        # in at least one of the input dicts
+        arg_keys = [
+            arg.keys()
+            for arg in (scale, direction, threshold)
+            if isinstance(arg, dict)
+        ]
+        cols = set(itertools.chain.from_iterable(arg_keys))
+
+    else:
+        # Append a flag col for each col in the dataframe
+        cols = df.columns
+
+    for col in sorted(cols):
+
+        # Allow arguments to be a mix of dict and single instances
+        s = scale.get(col, "percentage") if isinstance(scale, dict) else scale
+        d = (
+            direction.get(col, "any")
+            if isinstance(direction, dict)
+            else direction
+        )
+        t = (
+            threshold.get(col, 0.0)
+            if isinstance(threshold, dict)
+            else threshold
+        )
+
+        df[f"{col}_jump_flag"] = _flag_jumps_single_col(
```

Comment:
@loganthomas what do you think of using the `df.assign()` syntax here instead? That _might_ be more in-line with modern `pandas` syntax that the `pandas` devs are encouraging. This would look like:

```python
columns_to_add = {}
for column in columns:
    columns_to_add[column] = ...

df = df.assign(**columns_to_add)
```

Reply:
@ericmjl Wonderful! That makes perfect sense. 👍